engine/rowset/src/main/java/io/deephaven/engine/rowset/impl/singlerange/SingleRange.java

```diff
@@ -204,6 +204,14 @@ public final OrderedLongSet ixRemove(final long key) {
         return OrderedLongSet.twoRanges(rangeStart(), key - 1, key + 1, rangeEnd());
     }
 
+    private static long addSaturated(final long x, final long y) {
+        final long res = x + y;
+        if (res < x) {
+            return Long.MAX_VALUE;
+        }
```

Comment:
I doubt it matters - but we might be able to have a simpler check `res < 0`?

Reply:
We discussed over slack and we think it is better.
We can do a "math proof" that it will be < 0 if it overlaps based on the fact that there are more negative values (1 more) than positive values, and that we know x and y are both non-negative.